modules/frontend/searchsharding.go

```diff
@@ -182,7 +185,12 @@ func (s searchSharder) RoundTrip(r *http.Request) (*http.Response, error) {
 		startedReqs++
 
 		go func(innerR *http.Request) {
-			defer wg.Done()
+			defer func() {
+				if progress.shouldQuit() {
+					subCancel()
```

Comment:
What does this change do?  It looks like `subCancel` is already deferred up above, and the main for loop on line 174 will exit on the same criteria (`progress.shouldQuit() == true`)

Reply:
The searchResponse object/searchSharder used to cancel the context directly in `shouldQuit()`. I wanted to keep that responsibility in the main search sharding code so I moved it here.

Everytime a search sharding job completes this code checks to see if we should cancel all other jobs and does so if necessary. Reasons to cancel include things like a job failed or the results limit was reached.

Reply:
I'm still a little unclear.   I think it would have the same effect if not called here.   The outer loop on requests would break because `shouldQuit` is true, and then the method would exit and `subCancel()` is already deferred.